pkg/operator/deploymentcontroller/deployment_controller.go

```diff
@@ -303,6 +303,15 @@ func (c *DeploymentController) syncManaged(ctx context.Context, opSpec *opv1.Ope
 		status = status.WithConditions(progressingCondition)
 	}
 
+	// Degrade when operator is progressing too long.
+	progressingTooLong, err := v1helpers.IsUpdatingTooLong(opStatus, c.instanceName+opv1.OperatorStatusTypeProgressing)
```

Comment:
If this is the first `syncManaged`, `opStatus` has not yet populated progressing condition and thus `IsUpdatingTooLong` fails here. I must be missing something obvious, how it can pass?

Reply:
Good point, I probably had the status populated already during verification because there was an unpatched operator running before. If we go with the suggestion below to not return error here it will just appear as if it's not progressing in this case, which is better.